token/cli/src/main.rs

```diff
@@ -1423,35 +1457,73 @@ async fn command_close(
             )
             .into());
         }
-        config.check_owner(&account, &source_account.owner)?;
+    }
 
-        let recipient_account = config.rpc_client.get_token_account(&recipient).await?;
-        let is_recipient_wrapped = recipient_account.map(|x| x.is_native).unwrap_or(false);
-        (is_recipient_wrapped, source_account.owner)
-    };
+    let res = token
+        .close_account(&account, &recipient, &close_authority, &bulk_signers)
+        .await?;
 
-    let mut instructions = vec![close_account(
-        &program_id,
-        &account,
-        &recipient,
-        &close_authority,
-        &config.multisigner_pubkeys,
-    )?];
+    let tx_return = finish_tx(config, &res, false).await?;
+    Ok(match tx_return {
+        TransactionReturnData::CliSignature(signature) => {
+            config.output_format.formatted_string(&signature)
+        }
+        TransactionReturnData::CliSignOnlyData(sign_only_data) => {
+            config.output_format.formatted_string(&sign_only_data)
+        }
+    })
+}
+
+async fn command_close_mint(
+    config: &Config<'_>,
+    token_pubkey: Pubkey,
+    close_authority: Pubkey,
+    recipient: Pubkey,
+    bulk_signers: BulkSigners,
+) -> CommandResult {
+    let token = token_client_from_config(config, &token_pubkey);
+
+    if !config.sign_only {
+        let mint_account = config.rpc_client.get_account(&token_pubkey).await?;
```

Comment:
You'll need to update the program id in the config based on the owner of this account